src/Build/Graph/ProjectGraphNode.cs

```diff
@@ -21,11 +23,16 @@ internal ProjectGraphNode(Project project)
         /// <summary>
         /// Gets an unordered collection of graph nodes for projects which this project references.
         /// </summary>
-        public IReadOnlyCollection<ProjectGraphNode> ProjectReferences => Array.Empty<ProjectGraphNode>();
+        public IReadOnlyCollection<ProjectGraphNode> ProjectReferences => _projectReferences;
 
         /// <summary>
         /// Gets the evaluated project represented by this node in the graph.
         /// </summary>
         public Project Project { get; }
+
+        internal void AddProjectReferences(List<ProjectGraphNode> projectGraphNodes)
```

Comment:
Instead of an add call, can this be part of the ctor? It's always better to have it fully constructed when possible. Or I could also see an `AddProjectReference` if it were added immediately when discovered.

Reply:
I agree; I'd go so far as to move the reach-into-the-project-and-extract references code into this class and call it from the constructor.

Reply:
I have added an AddProjectReference call to update the parent node as soon as a child node is evaluated

Reply:
I wouldn't explore references in the constructor to make it easier to stub out the graph.